src/settings.rs

```diff
@@ -20,38 +20,45 @@ pub struct Settings {
 
 impl Default for Settings {
     fn default() -> Settings {
-        Settings {
-            debug: false,
-            port: DEFAULT_PORT,
-            database_url: "mysql://root@127.0.0.1/syncstorage".to_string(),
-            database_pool_max_size: None,
-            #[cfg(test)]
-            database_use_test_transactions: false,
-            master_secret: Secrets::default(),
-        }
+        Settings::with_env_and_config_file(None).unwrap()
     }
 }
 
 impl Settings {
-    /// Load the settings from the config file if supplied, then the environment.
-    pub fn with_env_and_config_file(filename: &Option<String>) -> Result<Self, ConfigError> {
-        let mut s = Config::default();
-        // Set our defaults, this can be fixed up drastically later after:
-        // https://github.com/mehcode/config-rs/issues/60
-        s.set_default("debug", false)?;
-        s.set_default("port", DEFAULT_PORT as i64)?;
-        #[cfg(test)]
-        s.set_default("database_use_test_transactions", false)?;
-        s.set_default("master_secret", "")?;
-
-        // Merge the config file if supplied
+    /// Construct a `Settings` instance, populating it with data from the file
+    /// system and local environment.
+    ///
+    /// Precedence (earlier items override later ones):
+    ///
+    ///   1. Environment variables: `$SYNC_<UPPERCASE_KEY_NAME>`
+    ///   2. File: `filename` argument
+    ///   3. File: `config/local.toml`
+    ///   4. File: `config/<$SYNC_ENV>.toml`
+    ///   5. File: `config/default.toml`
+    pub fn with_env_and_config_file(filename: Option<String>) -> Result<Self, ConfigError> {
+        let mut config = Config::new();
+
+        config.merge(File::with_name("config/default"))?;
```

Comment:
We don't need to include the .toml?

Reply:
Also, this means that the binary will no longer run at all unless you make a default config file in the right spot, right?

So we'll be going from a fairly pure '12-factor app' that requires no files in the docker and can have defaults tweaked with env vars, to a binary that requires files in the right spots. Is this what we want?

Reply:
No, the config crate tries TOML, JSON, YAML, HJSON and INI extensions automatically. I'll add a comment to that effect too.

Reply:
> So we'll be going from a fairly pure '12-factor app' that requires no files in the docker and can have defaults tweaked with env vars, to a binary that requires files in the right spots.

Everything can still be set with env vars fwiw, but...

> Is this what we want?

It sounds like the answer to that is no. I'm happy to close this.